libs/auth/src/signed_hash_validation.ts

```diff
@@ -1,95 +1,123 @@
+import { existsSync } from 'fs';
 import fs from 'fs/promises';
 import { Readable } from 'stream';
-import { assert } from '@votingworks/basics';
+import { assert, sleep } from '@votingworks/basics';
+import {
+  formatElectionHashes,
+  SignedHashValidationQrCodeValue,
+} from '@votingworks/types';
 
 import {
   constructSignedHashValidationConfig,
   SignedHashValidationConfig,
 } from './config';
 import { signMessage } from './cryptography';
-import { FileKey, TpmKey } from './keys';
+import { runCommand } from './shell';
 import { constructPrefixedMessage } from './signatures';
 
-const SIGNED_HASH_VALIDATION_MESSAGE_PAYLOAD_SEPARATOR = '/';
 const SIGNED_HASH_VALIDATION_QR_CODE_VALUE_SEPARATOR = ';';
+const SIGNED_HASH_VALIDATION_MESSAGE_PAYLOAD_SEPARATOR = '/';
+const SYSTEM_HASH_DISPLAY_LENGTH = 10;
```

Comment:
why are we truncating this hash? That makes things less secure, so wondering if we really need to.

Reply:
Yeah good q - I was just trying to avoid having to display such a long string in the UI. I would edit the below of course but gives you a sense of the design challenge:

<table>
<img width="1512" alt="display" src="https://github.com/user-attachments/assets/cd1efed5-07a6-4248-89e9-47660fc5f872">
</table>

I could of course sign the full hash but only display a truncated version. But that kinda defeats the purpose, I think.

I can come up with a design workaround.

Reply:
How about we base64-encode that hash, make it almost half the size? Then I think displaying it raw is okay.

Reply:
@benadida I like that - updated! https://github.com/votingworks/vxsuite/pull/5302/commits/1542496da162d486c421af4059b4f9df3dfb6436

![updated-screenshot](https://github.com/user-attachments/assets/94765e26-ffee-4c3e-bf5b-4a7e78ffe164)

Reply:
✅ PR description screenshot updated, too